server/src/main/java/org/elasticsearch/search/lookup/LeafFieldsLookup.java

```diff
@@ -129,31 +127,28 @@ public boolean containsValue(Object value) {
     private FieldLookup loadFieldData(String name) {
         FieldLookup data = cachedFieldData.get(name);
         if (data == null) {
-            MappedFieldType fieldType = mapperService.fieldType(name);
+            MappedFieldType fieldType = fieldTypeLookup.apply(name);
             if (fieldType == null) {
                 throw new IllegalArgumentException("No field found for [" + name + "] in mapping");
             }
             data = new FieldLookup(fieldType);
             cachedFieldData.put(name, data);
         }
         if (data.fields() == null) {
+            MappedFieldType fieldType = data.fieldType();
             List<Object> values;
-            if (TypeFieldType.NAME.equals(data.fieldType().name())) {
-                values = new ArrayList<>(1);
-                final DocumentMapper mapper = mapperService.documentMapper();
-                if (mapper != null) {
-                    values.add(mapper.type());
-                }
+            if (TypeFieldType.NAME.equals(fieldType.name())) {
+                values = Collections.singletonList(((TypeFieldType)fieldType).getType());
```

Comment:
this is not fantastic, yet it does the job... do we have better options? I like that this way we still only need the lookup function as argument.

Reply:
So I think we can actually remove this in master, as using the `_type` field in scripts is deprecated in 7.x.  But you might want to keep it for the backport, and then remove it later.

Reply:
I opened #64041